crates/cheatable-starknet/src/cheatcodes/deploy.rs

```diff
@@ -42,14 +42,16 @@ impl CheatedState {
         for felt in inputs.iter().skip(2).take(calldata_length) {
             calldata.push(felt.clone());
         }
-
+        // self.increment_deploy_counter();
+        // print!("deploy_counter {}", self.deploy_counter);
         // Deploy a contract using syscall deploy.
-        let account_address = ContractAddress(patricia_key!(TEST_ACCOUNT_CONTRACT_ADDRESS));
+        let account_address: ContractAddress =
+            ContractAddress(patricia_key!(TEST_ACCOUNT_CONTRACT_ADDRESS));
         let block_context = build_block_context();
         let entry_point_selector = selector_from_name("deploy_contract");
-        let salt = ContractAddressSalt::default();
+        let salt = self.gen_salt();
         let class_hash = ClassHash(StarkFelt::new(class_hash.to_be_bytes()).unwrap());
-
+        self.increment_deploy_counter();
```

Comment:
I wonder if we should increment it here or only after successful deploys
cc @Arcticae

Reply:
I think only after successful ones, but it doesn't matter that much (i guess?)